bdns/dns_test.go

```diff
@@ -190,13 +188,31 @@ func serveLoopResolver(stopChan chan bool) chan bool {
 			fmt.Println(err)
 		}
 	}()
-	return waitChan
+}
+
+func pollServer() {
+	var conn *dns.Conn
+	timeout := time.Duration(time.Millisecond * 200)
+	tries := 25
+
+	for i := 0; conn == nil; i++ {
```

Comment:
Instead of using `conn == nil` here and ` if i == tries` below this should just be a normal `i := 0; i < tries; i++` loop and the max tries test can be done outside the loop by checking if `conn == nil` and either printing an error and exiting or closing the existing connection.

Also instead of `fmt.Println` for printing the error message you should use `fmt.Fprintln(os.Stderr, ...)`, the message should also say that the tries were polling the test DNS server and the number of tries (i.e. `Max tries (25) reached while waiting for the test DNS server to come up`).